src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -1165,6 +1166,45 @@ - (MTRDeviceController * _Nullable)initializeController:(MTRDeviceController *)c
                                   error:error];
 }
 
+- (void)setMessageReliabilityProtocolIdleRetransmitMs:(nullable NSNumber *)idleRetransmitMs
+                                   activeRetransmitMs:(nullable NSNumber *)activeRetransmitMs
+                                    activeThresholdMs:(nullable NSNumber *)activeThresholdMs
+                          additionalRetransmitDelayMs:(nullable NSNumber *)additionalRetransmitDelayMs
+{
+    [self _assertCurrentQueueIsNotMatterQueue];
+    dispatch_sync(_chipWorkQueue, ^{
```

Comment:
Does this need to be a dispatch_sync?  The MRP params are not observable by any code that is not running on the Matter queue, after this change, right?  Message sending is on that queue, and so is our `GetLocalMRPConfig()` in this function after these changes.

So it seems to me that this can be an async dispatch just fine.